Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Allocator/Jobs/SendSearchRequestToDrivers/Handle/Internal/DriverPool.hs

```diff
@@ -399,23 +406,34 @@ prepareDriverPoolBatch driverPoolCfg searchReq searchTry tripQuoteDetails starti
                       }
                   )
                   opCityId
-              return $ filterSpecialDrivers specialDrivers calculateGoHomeDriverPoolBatch
+              let res' = filterSpecialDrivers specialDrivers calculateGoHomeDriverPoolBatch
+                  merchantId = searchReq.providerId
+                  isRental = isRentalTrip searchTry.tripCategory
+                  isInterCity = isInterCityTrip searchTry.tripCategory
+              res <- filterM (scheduledRideFilter currentSearchInfo merchantId opCityId isRental isInterCity transporterConfig) res'
```

Comment:
It's there in other files as well, did you check this filter is not repeated?

Reply:
it's for three cases of pooling  normal , currentlyOnRide and GoHome .